test/discovery_failure_test.py

```diff
@@ -1,82 +1,77 @@
-
-    def delete_broken_import_file(self):
-        files = [
-            self.broken_import_file_path,
-            # Also remove the .pyc that was created if the file was imported.
-            self.broken_import_file_path + 'c',
-        ]
-        for f in files:
-            try:
-                os.remove(f)
-            except OSError as exc:
-                _log.error("Could not remove broken import file %s: %r" % (f, exc))
-
-    @class_setup
-    def setup_import_file(self):
-        self.create_broken_import_file()
-
-    @class_teardown
-    def teardown_import_file(self):
-        self.delete_broken_import_file()
+        filename = self.broken_import_module + '.py'
+        with io.open(filename, 'w') as test_file:
+            test_file.write(self.broken_contents)
+
+    @T.setup_teardown
+    def create_temporary_directory(self):
+        self.tempdir = tempfile.mkdtemp()
+        cwd = os.getcwd()
+        os.chdir(self.tempdir)
+        self._write_broken_import_file()
+        try:
+            yield
+        finally:
+            os.chdir(cwd)
+            shutil.rmtree(self.tempdir)
 
 
 class DiscoveryFailureTestCase(BrokenImportTestCase):
     def test_discover_test_with_broken_import(self):
         """Insure that DiscoveryError is raised when a test which imports a
         non-existent module is discovered."""
-        try:
-            discovered_tests = test_discovery.discover(self.broken_import_module)
-            discovered_tests.next()
-        except DiscoveryError as exc:
-            assert_in('No module named non_existent_module', str(exc))
-        else:
-            assert False, 'Expected DiscoveryError.'
+
+        stdout, stderr = cmd_output(
+            'python', '-m', 'testify.test_program', self.broken_import_module,
+        )
+        T.assert_in('DISCOVERY FAILURE', stdout)
+        T.assert_in('ImportError: No module named non_existent_module', stderr)
 
 
 class DiscoveryFailureUnknownErrorTestCase(BrokenImportTestCase):
-    @class_setup
-    def setup_import_file(self):
-        self.create_broken_import_file(contents='raise AttributeError("aaaaa!")')
+    broken_contents = 'raise AttributeError("aaaaa!")\n'
 
     def test_discover_test_with_unknown_import_error(self):
         """Insure that DiscoveryError is raised when a test which raises an unusual exception upon import is discovered."""
-
-        try:
-            discovered_tests = test_discovery.discover(self.broken_import_module)
-            discovered_tests.next()
-        except DiscoveryError as exc:
-            assert_in('Got unknown error when trying to import', str(exc))
-        else:
-            assert False, 'Expected DiscoveryError.'
+        stdout, stderr = cmd_output(
+            'python', '-m', 'testify.test_program', self.broken_import_module,
+        )
+        T.assert_in('DISCOVERY FAILURE', stdout)
```

Comment:
I'd really rather see the whole string assert_equal'd, unless there's too many bits that vary.